conanfile.py

```diff
@@ -95,21 +95,21 @@ def package_info(self):
 
         linker_path = os.path.join(self.package_folder, "linkers")
 
-        self.cpp_info.set_property("cmake_file_name", "libhal-stm32f10x")
+        self.cpp_info.set_property("cmake_file_name", "libhal-stm32")
         self.cpp_info.set_property("cmake_find_mode", "both")
 
-        self.cpp_info.components["stm32f10x"].set_property(
-            "cmake_target_name",  "libhal::stm32f10x")
-        self.cpp_info.components["stm32f10x"].exelinkflags.append(
+        self.cpp_info.components["stm32"].set_property(
+            "cmake_target_name",  "libhal::stm32")
+        self.cpp_info.components["stm32"].exelinkflags.append(
             "-L" + linker_path)
-        self.cpp_info.components["stm32f10x"].requires = requirements_list
+        self.cpp_info.components["stm32"].requires = requirements_list
 
         def create_component(self, component, flags):
-            link_script = "-Tlibhal-stm32f10x/" + component + ".ld"
+            link_script = "-Tlibhal-stm32/" + component + ".ld"
             component_name = "libhal::" + component
             self.cpp_info.components[component].set_property(
                 "cmake_target_name", component_name)
-            self.cpp_info.components[component].requires = ["stm32f10x"]
```

Comment:
These should actually stay the same. We still want to reference the specific stm32f series chip by its name/subseries.